crates/sui-core/src/transaction_input_checker.rs

```diff
@@ -247,6 +263,10 @@ mod checked {
         let mut all_objects = Vec::with_capacity(input_objects.len());
 
         for (object_kind, object) in input_objects.into_iter().zip(objects) {
+            if deleted_shared_objects.contains_key(&object.id()) {
```

Comment:
I think I understand whats going on here, but a comment about this always being empty at signing or something might help (or maybe I don't understand whats going on here in which case we really need a comment :P)